closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1053 @@
+                }
+
+                if (debug != null) {
+                    debug.println("Restricted security mode loaded.");
+                    debug.println("Restricted security mode properties: " + props.toString());
+                }
+
+                securityEnabled = true;
+            }
+
+        } catch (Exception e) {
+            if (debug != null) {
+                debug.println("Unable to load restricted security mode configurations.");
+            }
+            printStackTraceAndExit(e);
+        }
+        return securityEnabled;
+    }
+
+    /**
+     * Load user restricted security settings from system property.
+     */
+    private static void initUserSetting() {
+        if (debug != null) {
+            debug.println("Loading user restricted security settings.");
+        }
+
+        String[] inputs = userSecuritySetting.split(",");
+
+        // For input ",,"
+        if (inputs.length == 0) {
+            printStackTraceAndExit("User restricted security setting " + userSecuritySetting + " incorrect.");
+        }
+
+        for (String input : inputs) {
+            String in = input.trim();
+            if (in.equalsIgnoreCase("audit")) {
+                userSecurityAudit = true;
+            } else if (in.equalsIgnoreCase("help")) {
+                userSecurityHelp = true;
+            } else if (in.equalsIgnoreCase("trace")) {
+                userSecurityTrace = true;
+            } else {
+                try {
+                    userSecurityNum = Integer.parseInt(in);
+                } catch (NumberFormatException e) {
+                    printStackTraceAndExit("User restricted security setting " + userSecuritySetting + " incorrect.");
+                }
+            }
+        }
+
+        if (debug != null) {
+            debug.println("Loaded user restricted security settings, with userSecurityNum: " + userSecurityNum
+                    + " userSecurityTrace: " + userSecurityTrace
+                    + " userSecurityAudit: " + userSecurityAudit
+                    + " userSecurityHelp: " + userSecurityHelp);
+        }
+    }
+
+    /**
+     * Add restricted security providers.
+     *
+     * @param providers the provider name array
+     */
+    private static void setProviders(Properties props, List<String> providers) {
+        if (debug != null) {
+            debug.println("Adding restricted security provider.");
+        }
+
+        int pNum = 0;
+        for (String provider : providers) {
+            pNum += 1;
+            props.setProperty("security.provider." + pNum, provider);
+            if (debug != null) {
+                debug.println("Added restricted security provider: " + provider);
+            }
+        }
+    }
+
+    /**
+     * Add restricted security properties.
+     *
+     * @param props the java.security properties
+     */
+    private static void setProperties(Properties props, RestrictedSecurityProperties properties) {
```

Comment:
Suggest removing `RestrictedSecurityProperties properties` and using `restricts` instead.